### PR TITLE
Openstack Blockstore and Object store plugins

### PR DESCRIPTION
**Pull request by drekle**
The example app will create the following data in swift:
```
ark create backup demo -l app=nginx --include-resources=* -w
```
```
$ openstack container show backups
+--------------+---------+
| Field        | Value   |
+--------------+---------+
| account      | v1      |
| bytes_used   | 7071    |
| container    | backups |
| object_count | 5       |
+--------------+---------+
```
| name | size |
|---|---|
| ark-backup.json | 1.17 KB |
| demo-logs.gz | 2.73 KB | 
| demo-volumesnapshots.json.gz | 274 bytes | 
| demo.tar.gz | 2.71 KB |

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,78 @@
...
+
+Backup storage:
+
+1. Run `kubectl edit backupstoragelocation <location-name> -n <velero-namespace>` e.g. `kubectl edit backupstoragelocation default -n velero` OR `velero backup-location create <location-name> --provider openstack`
+2. Change the value of `spec.provider` to openstack to enable an **Object Store** plugin
+3. Save and quit. The plugin will be used for the next `backup/restore`
+
+Volume snapshot storage:
+
+1. Run `kubectl edit volumesnapshotlocation <location-name> -n <velero-namespace>` e.g. `kubectl edit volumesnapshotlocation default -n velero` OR `velero snapshot-location create <location-name> --provider openstack`
```
Comment (nrb) [inline README.md]: Should the provider here be `openstack-blockstore`, as in the YAML examples?

File: examples/05-ark-backupstoragelocation.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+# DEPRECATED
+
+apiVersion: ark.heptio.com/v1
+kind: BackupStorageLocation
+metadata:
+  name: default
+  namespace: heptio-ark
+spec:
+  provider: openstack
```
Comment (nrb) [inline examples/05-ark-backupstoragelocation.yaml:23]: Should this reference be `openstack-objectstore`?

File: README.md (hunk 1 above)
Comment (drekle) [reply README.md]: Yes thanks, I forgot to update probably a good chunk of the readme.

**Comment by drekle** [comment]
I'm going to retest after the last refactor.

**Comment by drekle** [comment]
Going to merge as we likely will not use velero.

**Pull request merged**